### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by damencho**
This pull request includes several updates to the Matroska file format handling in the `src/main/java/org/ebml/matroska` directory. The changes primarily involve the addition and removal of elements in the `MatroskaDocType` enum and the corresponding `MatroskaDocTypes` class, as well as updates to methods for writing and updating Matroska file elements.

### CODE REVIEW

File: src/main/java/org/ebml/matroska/MatroskaFileWriter.java [hunk 1]
```diff
@@ -254,14 +254,15 @@ public void addFrame(final MatroskaFileFrame frame)
...
       flush();
-
-      if (ioDW.isSeekable())
-      {
-        final long clusterPos = ioDW.getFilePointer();
-        cueData.addCue(clusterPos, frame.getTimecode(), frame.getTrackNo());
-      }
     }
+
+    boolean addCue = !cluster.getTracks().contains(frame.getTrackNo());
```
Comment (bgrozev) [inline src/main/java/org/ebml/matroska/MatroskaFileWriter.java:246]: The `isSeekable` check was removed

File: src/main/java/org/ebml/matroska/MatroskaFileTrack.java [hunk 2]
```diff
@@ -53,7 +53,7 @@ public class MatroskaFileTrack
   private boolean flagEnabled = true;
   private boolean flagDefault = true;
   private boolean flagForced = false;
-  private boolean flagLacing = true;
+  private boolean flagLacing = false;
```
Comment (bgrozev) [inline src/main/java/org/ebml/matroska/MatroskaFileTrack.java]: I think this defaults to `true` in the spec, and it seems like a safe default (the track MAY contain blocks using lacing, while `false` asserts that it doesn't). Why change it? If we want it `false` for our case we should probably set it in jmr

File: src/main/java/org/ebml/matroska/MatroskaFileWriter.java [hunk 3]
```diff
@@ -53,6 +54,8 @@ public class MatroskaFileWriter implements Closeable
 
   private long clusterLen = 0;
 
+  private long lastDuration = 0;
```
Comment (bgrozev) [inline src/main/java/org/ebml/matroska/MatroskaFileWriter.java]: Suggest renaming to `duration` or `segmentDuration`

File: src/main/java/org/ebml/matroska/MatroskaFileWriter.java [hunk 4]
```diff
@@ -246,6 +249,16 @@ public void addFrame(final MatroskaFileFrame frame)
...
+
+    long defaultDuration = 0;
+
+    MatroskaFileTrack tr = tracks.getTrack(frame.getTrackNo());
+    if (tr != null)
+    {
+      defaultDuration = tr.getDefaultDuration();
+    }
+
+    lastDuration = cluster.getLastTimecode() + TimeUnit.NANOSECONDS.toMillis(defaultDuration);;
```
Comment (bgrozev) [inline src/main/java/org/ebml/matroska/MatroskaFileWriter.java]: This assumes that the first timecode is 0, and that frames from all tracks are always added in order (e.g. addFrame(track=1, timecode=1000); addFrame(track=2, timecode=900) is handled incorrectly and we may end up using something like that after buffering in jmr). 

I suggest just saving the smallest and largest timecode of all frames added to the segment, then setting duration to the difference. `cluster.getLastTimecode()` is always just `frame.getTimecode()`, so no need to expose it

File: src/main/java/org/ebml/matroska/MatroskaFileTrack.java (hunk 2 above)
Comment (damencho) [reply src/main/java/org/ebml/matroska/MatroskaFileTrack.java]: 👍 will drop the commit.

File: src/main/java/org/ebml/matroska/MatroskaFileWriter.java (hunk 4 above)
Comment (damencho) [reply src/main/java/org/ebml/matroska/MatroskaFileWriter.java]: Changed that, thanks.

File: src/main/java/org/ebml/matroska/MatroskaFileWriter.java (hunk 1 above)
Comment (damencho) [reply src/main/java/org/ebml/matroska/MatroskaFileWriter.java:246]: fixed it.

**Review by bgrozev — APPROVED** [review]

**Pull request merged**